### PR TITLE
fix(#1005): Rewritten organization management page from JSP to React JS

### PR DESCRIPTION
**Pull request by adityadeshlahre**
fixes(#1005)
## NOTE : will paste the video of workFlow later on
- restController routes need to test it more
- UI need to tested more
- dataFetching for modify component need to be me more robust

# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the Issue number if applicable.
- [x] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3 Styleguide and design documentation.
- [x] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing Guidelines of this project.

### main page
image

### add & modify org page [90% of its working and UI same]
image

[Add relevant screenshots here if applicable]

## Related Issue

[Add a link to the related issue or mention it here if applicable]

## Other

[Add any additional information or notes here]

### CODE REVIEW

**Comment by adityadeshlahre** [comment]
@mozzy11 @gitcliff 

# workFlow video on REACT UI of Organization Management Page

workFlowOM

#### things left
- Post requests
- Save button fixes
- sever more testing
- few edge case need to fix

## NOTE : IGNORE THOSE DEBUGGING BUTTONS, FOR NOW, WILL CLEAN IT UP LATER ON

Thank You :slightly_smiling_face: 
If any changes required please let me know !

**Comment by gitcliff** [comment]
thanks @adityadeshlahre for the great work,,
would like to inquire on where the `Previous` and `Next` buttons are to switch from one page to another plus am also not seeing the `CLIA Number` columns as seen in the screenshot
Screenshot 2024-05-13 at 3 57 51 PM

**Comment by adityadeshlahre** [comment]
> thanks @adityadeshlahre for the great work,, would like to inquire on where the `Previous` and `Next` buttons are to switch from one page to another plus am also not seeing the `CLIA Number` columns as seen in the screenshot Screenshot 2024-05-13 at 3 57 51 PM


### these are the next & previous buttons to navigate through list!
image

### there CLIA number column
do you see the scroller bar at the lower part of video? 
if you scroll through u will see it 
By mistake, screenSize was too small that on recording that column did appear :slightly_smiling_face: 
image

**Comment by mherman22** [comment]
> thanks @adityadeshlahre for the great work,, would like to inquire on where the `Previous` and `Next` buttons are to switch from one page to another plus am also not seeing the `CLIA Number` columns as seen in the screenshot Screenshot 2024-05-13 at 3 57 51 PM

@adityadeshlahre, In addition to Cliff's question here. The rest controller implements paging. What that means is that it only sends 20 items on request which I guess are the ones we are able to see in the data table currently. However if you notice, the old ui shows that we have about 1000+ items in the db.

Meaning we've to improving the pagination we currently have. So that when we click next page, we are sending something like `organization?paging=2` in order to fetch the next page of 20 items to display in the data table🙂

File: src/main/java/org/openelisglobal/organization/controller/rest/OrganizationRestController.java [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+
+        Organization organization;
+
+        // redirect to get organization for next or previous entry
+        if (FWD_NEXT.equals(request.getParameter("direction"))) {
+            organization = organizationService.getNext(id);
+            String newId = organization.getId();
+
+            // return new ModelAndView("redirect:/Organization?ID=" + Encode.forUriComponent(newId) + "&startingRecNo="
+            //         + Encode.forUriComponent(start));
```
[the remark this answers is not in the document]
Comment (adityadeshlahre) [reply src/main/java/org/openelisglobal/organization/controller/rest/OrganizationRestController.java:177]: i will clean up after wards

**Comment by adityadeshlahre** [comment]
> > thanks @adityadeshlahre for the great work,, would like to inquire on where the `Previous` and `Next` buttons are to switch from one page to another plus am also not seeing the `CLIA Number` columns as seen in the screenshot Screenshot 2024-05-13 at 3 57 51 PM
> 
> @adityadeshlahre, In addition to Cliff's question here. The rest controller implements paging. What that means is that it only sends 20 items on request which I guess are the ones we are able to see in the data table currently. However if you notice, the old ui shows that we have about 1000+ items in the db.
> 
> Meaning we've to improve the pagination we currently have. So that when we click next page, we are sending something like `organization?paging=2` in order to fetch the next page of 20 items to display in the data table🙂

Yes @mherman22 I know this is the edge case and will fix that in further commits.

**Comment by mozzy11** [comment]
Thanks @adityadeshlahre . any blocker or progress ??

**Comment by adityadeshlahre** [comment]
### Organization Management WorkFlow

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/132184385/6de42938-b182-463c-bfb8-3b646be5a7f4

If any edge cases are found let me know :slightly_smiling_face: 
else this PR is ready to merge 

Thank You 

CC @mozzy11 & @gitcliff

**Comment by gitcliff** [comment]
thanks @adityadeshlahre,,just one more issue ....
could you resolve the  
`Publish 3.X FrontEnd Docker image / build-and-push-image (pull_request)` build failure , mya be you need to update the branch

**Review by gitcliff — APPROVED** [review]

**Pull request merged**